rules/Php74/Rector/Property/TypedPropertyRector.php

```diff
@@ -259,12 +238,17 @@ private function shouldSkipProperty(Property $property): bool
             return true;
         }
 
-        if (! $this->privatePropertyOnly) {
+        if ($property->isPrivate()) {
             return $this->propertyAnalyzer->hasForbiddenType($property);
         }
 
-        if ($property->isPrivate()) {
-            return $this->propertyAnalyzer->hasForbiddenType($property);
+        if ($property->isProtected()) {
+            $classReflection = $scope->getClassReflection();
+
+            // is we're in final class, the type can be changed
+            if ($classReflection instanceof ClassReflection && $classReflection->isFinal()) {
```

Comment:
it may cause error when class extends another class, eg:

```php
class A
{
    protected $data;
}

final class B extends A
{
    protected array $data;
    
    public function run()
    {
        $this->data = [];
        var_dump($this->data);
    }
}

(new B())->run();
```

that will cause error:

```bash
Fatal error: Type of B::$data must not be defined (as in class A) 
```

ref https://3v4l.org/PYOrX

Reply:
How should we handle protected properties then?

Reply:
I think should be only proceed when final and no extends:

```suggestion
            if ($classReflection instanceof ClassReflection && $classReflection->isFinal() && $classReflection->getParents() === []) {
```

Reply:
Thanks :+1: 

I'll add a test case for this one